### PR TITLE
Add public getter for instanceType

### PR DESCRIPTION
**Pull request by Jimilian**
It would allow to get information about instance type from groovy scripts

### CODE REVIEW

**Comment by Jimilian** [comment]
@francisu, can you take a look, please?

**Comment by thoulen** [comment]
Potential can increase the number of the call to the AWS API

File: src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java [hunk 1]
```diff
@@ -152,6 +153,7 @@ public EC2AbstractSlave(String name, String instanceId, String description, Stri
         this.launchTimeout = launchTimeout;
         this.amiType = amiType;
         readResolve();
+        fetchLiveInstanceData(true);
```
[the remark this answers is not in the document]
Comment (Jimilian) [reply src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java]: It's a funny story. Jenkins allows to access object fields from groovy code. Also most of this fields are public and any of them can be used outside of this plugin.
Accessing fields directly gives unique opportunity to reduce the load and latency by not requesting AWS. To make this behavior possible somebody should initialize all this properties first.

**Comment by Jimilian** [comment]
`fetchLiveInstanceData`will be called pretty soon anyway. The calling it in constructor only removes potential "race condition" when somebody smart called object field before it was initialized by `fetchLiveInstanceData`.
Also keep in mind that `fetchLiveInstanceData` throttles number of queries.

File: src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java [hunk 2]
```diff
@@ -560,6 +564,11 @@ public String getPrivateDNS() {
         return privateDNS;
     }
 
+    public String getInstanceType() {
+        fetchLiveInstanceData(false);
+        return instanceType;
+    }
```
Comment (batmat) [inline src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java]: I do not think this is needed if the goal is for getting it from Groovy scripts.

File: src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java [hunk 3]
```diff
@@ -98,6 +98,7 @@
     public List<EC2Tag> tags;
     public final String cloudName;
     public AMITypeData amiType;
+    private String instanceType = null;
```
Comment (batmat) [inline src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java]: Per consistency with the above, and readability, initializing variables in Java is unneeded FWIW. What's written above is what happens anyway.

File: src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java (hunk 2 above)
Comment (Jimilian) [reply src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java]: I think it's better to have it - mainly for the consistency and to make it exposable for other plugins.

**Pull request merged**